flink-sql-runner/src/test/java/com/github/streamshub/flink/SqlRunnerTest.java

```diff
@@ -0,0 +1,42 @@
+package com.github.streamshub.flink;
+
+import java.util.List;
+import org.junit.jupiter.api.Test;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+class SqlRunnerTest {
```

Comment:
Another behaviour to test is that `parseStatements` adds a trailing `;` if none is present.

Maybe something for another PR, but I'd refactor out the work in the SqlRunner main method to make it testable. We have this untested logic where set statements are put in the configuration of the table env rather than executed. You could separate the environment/execution logic from the statement processing to make it more testable.

Reply:
That's a good point. Given that we already diverged off a little bit from the upstream, I think we should refactor it to make it more testable.

Reply:
I have added a test case for adding trailing semicolon to the end of the statements. I have left the refactor for another PR.

Reply:
Is it worth seeing if upstream would be interested in the testing changes?

Reply:
Yes, potentially. I think because it's an example code, they didn't have any tests?